tree/ntuple/v7/test/rfield_unsplit.cxx

```diff
@@ -120,3 +120,39 @@ TEST(RField, UnsupportedUnsplit)
    auto model = RNTupleModel::Create();
    EXPECT_THROW(model->MakeField<CustomStreamerForceUnsplit>("f"), RException);
 }
+
+TEST(RField, UnsplitPoly)
+{
+   FileRaii fileGuard("test_ntuple_rfield_unsplit_poly.root");
+   {
+      auto model = RNTupleModel::Create();
+      model->AddField(RFieldBase::Create("p", "PolyContainer").Unwrap());
+      auto writer = RNTupleWriter::Recreate(std::move(model), "ntpl", fileGuard.GetPath());
+      auto ptrPoly = writer->GetModel().GetDefaultEntry().GetPtr<PolyContainer>("p");
+      ptrPoly->fPoly = std::make_unique<PolyBase>();
+      ptrPoly->fPoly->x = 0;
+      writer->Fill();
+      ptrPoly->fPoly = std::make_unique<PolyA>();
+      ptrPoly->fPoly->x = 1;
+      dynamic_cast<PolyA *>(ptrPoly->fPoly.get())->a = 100;
+      writer->Fill();
+      ptrPoly->fPoly = std::make_unique<PolyB>();
+      ptrPoly->fPoly->x = 2;
+      dynamic_cast<PolyB *>(ptrPoly->fPoly.get())->b = 200;
+      writer->Fill();
+   }
+
+   auto reader = RNTupleReader::Open("ntpl", fileGuard.GetPath());
```

Comment:
Since the reader and writer are in the same process and the data type does not change schema, this does not yet test whether the StreamerInfo is stored and restored correctly.